From 7e796793aa45752d93566099f5e31ed1391bf8c0 Mon Sep 17 00:00:00 2001
From: "Eric S. Raymond" <esr@thyrsus.com>
Date: Tue, 1 May 2012 18:11:25 -0400
Subject: [PATCH] Allow use of the CTS line for NNTP.

Nathan Knotts wrote: "I'm currently using the Xtreme104 12 port serial
board which only has the CTS handshaking lines exposed. Previous gpsd
releases had support for the cts handshaking pin via compile time
options, then later, command line options. As of release 3.5, these
options have been removed and automatic detection occurs.  However,
the bitmask defining which control lines to listen to (PPS_LINE_TIOC)
does not include the CTS line. I've added TIOCM_CTS to that bitmask to
include CTS capability. This has been tested to work with a ublox chip
outputting NMEA."
---
 ntpshm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/ntpshm.c
+++ b/ntpshm.c
@@ -623,7 +623,7 @@ static /*@null@*/ void *gpsd_ppsmonitor(
 
     /* root privileges are not required after this point */
 
-#define PPS_LINE_TIOC (TIOCM_CD|TIOCM_CAR|TIOCM_RI)
+#define PPS_LINE_TIOC (TIOCM_CD|TIOCM_CAR|TIOCM_RI|TIOCM_CTS)
 
     /* 
      * Wait for status change on any handshake line. The only assumption here 
